### ISSUE TITLE
Invalid options produce weird error messages

### ISSUE THREAD

**Issue by pschlump** [xenolf/lego#77]
Example:
```
   $ ./lego --email="pschlump@test1.com" --server="http://192.168.0.200:4000/directory" --http --domains="test1.com" run

```

Outputs

```
2016/01/17 18:42:46 Please specify --domains or -d
```

The real problem is that --http should be --http=":80" but the error is ununsual.

If you want me to fix this I can build a "pull" request.

**Comment by xenolf** [comment]
This is quite weird, I agree.
I guess this comes from the fact that the `--domains` switch is one of the two switches we validate for being there.
I'd thankfully accept a PR for that. It might take some time for me to get to it.

**Comment by pschlump** [comment]
Will do

On Mon, Jan 18, 2016 at 2:12 PM, xenolf <notifications@github.com> wrote:

> This is quite weird, I agree.
> I guess this comes from the fact that the --domains switch is one of the
> two switches we validate for being there.
> I'd thankfully accept a PR for that. It might take some time for me to get
> to it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xenolf/lego/issues/77#issuecomment-172655096>.
>



-- 
Philip Schlump

**Comment by xi2** [comment]
I think this may have been fixed by c3abd54? The command

```
$ ./lego --email="pschlump@test1.com" --server=https://acme-staging.api.letsencrypt.org/directory --http --domains="test1.com" run
```

Now returns

```
2016/03/17 22:49:23 The --http switch only accepts interface:port or :port for its argument.
```

**Comment by xenolf** [comment]
Yes, the mentioned commit should indeed fix this as the http/tls switches are checked before the domains are used.
